File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/Delivery.cs
namespace OctoSupply.Api.Models;

public sealed record Delivery(
    int DeliveryId,
    int SupplierId,
    string DeliveryDate,
    string Name,
    string? Description,
    string Status);

public sealed class CreateDeliveryRequest
{
    public int SupplierId { get; set; }
    public string DeliveryDate { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Status { get; set; } = "pending";
}

public sealed class UpdateDeliveryRequest
{
    public int? SupplierId { get; set; }
    public string? DeliveryDate { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public string? Status { get; set; }
}

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/delivery.py
from __future__ import annotations

from pydantic import BaseModel


class Delivery(BaseModel):
    deliveryId: int
    supplierId: int
    deliveryDate: str
    name: str
    description: str | None = None
    status: str


class CreateDeliveryRequest(BaseModel):
    supplierId: int
    deliveryDate: str
    name: str
    description: str | None = None
    status: str = "pending"


class UpdateDeliveryRequest(BaseModel):
    supplierId: int | None = None
    deliveryDate: str | None = None
    name: str | None = None
    description: str | None = None
    status: str | None = None


class UpdateDeliveryStatusRequest(BaseModel):
    status: str
    deliveryPartner: str | None = None
